### PR TITLE
[FIX] Black screen during Share

### PR DESCRIPTION
**Pull request by cprice-kgi**
## Proposed changes
Something about RNSplashScreen.hide is causing an issue here. I wrapped the call in a try/catch.

## Issue(s)	
#3088

## How to test or reproduce
Try to share an image either from within the app or from an another application to the app. After investigation, I have found this issue appears only when the server has the mobile screen lock value enabled. If it is disabled, sharing will work fine. If enabled, the call to RNSplashScreen.hide will cause a black screen to appear, and the user is unable to proceed.

## Types of changes

- [ x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x ] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [ x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
N/A

### CODE REVIEW

**Review by LevyCosta — APPROVED** [review]
lgtm

File: app/utils/localAuthentication.js [hunk 1]
```diff
@@ -103,7 +104,11 @@ export const localAuthenticate = async(server) => {
 	// if screen lock is enabled
 	if (serverRecord?.autoLock) {
 		// Make sure splash screen has been hidden
-		await RNBootSplash.hide();
+		try {
+			await RNBootSplash.hide();
+		} catch (error) {
+			logEvent(events.LC_SERVER_RECORD_BOOT_SPLASH_HIDE_ERROR);
```
Comment (diegolmello) [inline app/utils/localAuthentication.js:110]: @reinaldonetof This is wrong imo.
If we want to log something, let's log the error.

File: app/views/AttachmentView.js [hunk 2]
```diff
@@ -106,9 +106,7 @@ class AttachmentView extends React.Component {
 	handleSave = async () => {
 		const { attachment } = this.state;
 		const { user, baseUrl } = this.props;
-		const {
-			title_link, image_url, image_type, video_url, video_type
-		} = attachment;
+		const { title_link, image_url, image_type, video_url, video_type } = attachment;
```
Comment (diegolmello) [inline app/views/AttachmentView.js:109]: Unrelated, but prettier asked me to change.
It won't hurt :)

**Comment by diegolmello** [comment]
The issue is that we're calling `hide` without having a splash screen on share extension.
That triggers `react-native-bootsplash has not been initialized`, similar to https://github.com/zoontek/react-native-bootsplash/issues/264
Recently they wrapped the `hide` call in a `try/catch` in the example https://github.com/zoontek/react-native-bootsplash/commit/41e32e8965b8808a3d0db5a32a56c4228fdab863#diff-f92b617bbc90bafc01461177b82175f320578d91293fddfbc810bacb339898edR20

**Comment by diegolmello** [comment]
@cprice-kgi Merged. Thanks for your contribution :)

**Comment by jsalatiel** [comment]
Is this the same problem that happens when one tries to share a audio received on whatsapp with someone on rockechat ? After choosing the contact for share, rocketchat gets  stuck on a blank window.  ( on android , running version 4.20)

**Pull request merged**